modules/dbparser/tests/test_patternize.c

```diff
@@ -334,25 +399,12 @@ testcase_find_clusters_slct(gchar *logs, guint support, gchar *expected)
       find_data->lines = expected_lines;
       find_data->num_of_lines = num_of_expected_lines;
       find_data->logs = logmessages->logmessages;
-      test_cluster = (Cluster *) g_hash_table_find(clusters, test_clusters_find, find_data);
-
-      if (!test_cluster || test_cluster->loglines->len != expected_support)
-        {
-          if (!test_cluster)
-            fprintf(stderr, "No cluster found;");
-          else
-            fprintf(stderr, "Support value does not match;");
+      test_cluster = (Cluster *) g_hash_table_find(clusters, _clusters_find, find_data);
 
-          fprintf(stderr, " expected_cluster='%s', expected_support='%d'\n", expected_item[0], expected_support);
-          fprintf(stderr, "Input:\n%s\n", logs);
-          fprintf(stderr, "Got clusters:\n");
-          g_hash_table_foreach(clusters, _debug_print2, NULL);
-
-          fail = TRUE;
-        }
+      cr_expect_not(!test_cluster || test_cluster->loglines->len != expected_support,
+                    "expected_cluster='%s', expected_support='%d'\nInput:\n%s\n", expected_item[0], expected_support, param->logs);
 
       g_free(find_data);
-//      g_strfreev(expected_line_strings); // FIXME: this segfaults, so let's leak it instead :)
```

Comment:
Removing this line do not really solve the original problem, I would either won't touch it or try resolving.

Reply:
expected_line_strings variable doesn't exist in updated code. as far as I see any other dynamicaly allocated elemets are cleaned up. let me know if I missunderstand something.

Reply:
Yes, you're right my bad.